### ISSUE TITLE
Provide possibility to create Grafana OnCall token trough terrafrom

### ISSUE THREAD

**Issue by Hronom** [grafana/terraform-provider-grafana#995]
Right now [Terraform scripts documentation recommends](https://registry.terraform.io/providers/grafana/grafana/latest/docs#managing-grafana-oncall) to go into the page of Grafana OnCall plugin and generate token.

Please provide possibility to create this token using Terraform scripts [as it done](https://registry.terraform.io/providers/grafana/grafana/latest/docs/resources/cloud_access_policy_token) with `grafana_cloud_access_policy_token`

**Comment by julienduchesne** [comment]
https://github.com/grafana/oncall/pull/3189 looks promising for this. It's still early (alpha as mentioned in the PR) but it sounds like it could work for TF

**Comment by fatihyildiz-cs** [comment]
Hey @julienduchesne  , did you get it working?

**Comment by julienduchesne** [comment]
Haven't tried it out. Just keeping track of options

**Comment by RoopeHakulinen** [comment]
@julienduchesne just to clarify, the PR you linked only allows using a service account for the resolution notes? 

Correct me if I'm wrong but the lack of this feature makes it not possible to create Grafana and Grafana OnCall as part of a single Terraform stack (without manual intervention)? That seems like a pretty important use case and is blocking us right now.

**Comment by julienduchesne** [comment]
You are not wrong, it's not possible to use it right now. But it's also not possible to programatically create an oncall token right now. It's promising because it may be the way forward in the future, when/if it is extended to all API endpoints. I'm just keeping notes in the open

**Comment by RoopeHakulinen** [comment]
Thanks, appreciate the quick answer. So am I right to assume that I should raise this (ability to create tokens programmatically) as an issue with the OnCall team?

**Comment by julienduchesne** [comment]
You may, though the service account seem like the way forward that they've decided and they're already working on it

**Comment by matiasb** [comment]
FYI, it is now possible to manage OnCall resources authenticating the provider via a [service account token](https://grafana.com/docs/grafana-cloud/developer-resources/infrastructure-as-code/terraform/terraform-oncall/#authentication-via-service-account-token).

**Comment by Duologic** [comment]
Closing this as per comment above.
